csrc/index_compute.cpp

```diff
@@ -1222,6 +1222,8 @@ Val* getHaloExtentOfRootAxis(IterDomain* id, Val* normal_extent = nullptr) {
         SimplifyingIrBuilder::create<Val>(
             (int64_t)halo.width(), DataType::Index));
     return halo_extent;
+  } else if (id->isDeviceDim()) {
```

Comment:
just out of curiosity: when is "Halo" involved and what does it mean?

Reply:
An explanation would also be helpful for me. I've seen halo mentioned in papers before...
@naoyam is it the same halo exchange for data format as mentioned in the gspmd paper? (https://arxiv.org/pdf/2105.04663.pdf)

Reply:
Halo is just a common word in parallel scientific computing. It's a boundary dependent region when a problem is partitioned on disjoint memory spaces. https://wgropp.cs.illinois.edu/courses/cs598-s15/lectures/lecture25.pdf

Reply:
No need to worry about the current halo support. I'll most likely remove it soon as it should be completely redesigned from scratch.